nixos/modules/services/monitoring/certspotter.nix

```diff
@@ -0,0 +1,126 @@
+{ config
+, lib
+, pkgs
+, ... }:
+
+let
+  cfg = config.services.certspotter;
+
+  configDir = pkgs.linkFarm "certspotter-config"
+    (lib.toList {
+      name = "watchlist";
+      path = pkgs.writeText "cerspotter-watchlist" (builtins.concatStringsSep "\n" cfg.watchlist);
+    }
+    ++ lib.optional (cfg.emailRecipients != [ ]) {
+      name = "email_recipients";
+      path = pkgs.writeText "cerspotter-email_recipients" (builtins.concatStringsSep "\n" cfg.emailRecipients);
+    }
+    ++ lib.optional (cfg.hooks != [ ]) {
+      name = "hooks.d";
+      path = pkgs.linkFarm "certspotter-hooks" (lib.imap1 (i: path: {
+        inherit path;
+        name = "hook${toString i}";
+      }) cfg.hooks);
+    });
+in
+{
+  options.services.certspotter = {
+    enable = lib.mkEnableOption "Cert Spotter, a Certificate Transparency log monitor";
+
+    sendmailPath = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        Path to the `sendmail` binary. By default, the local sendmail wrapper is used
+        (see `config.services.mail.sendmailSetuidWrapper`).
+      '';
+      example = lib.literalExpression ''"''${pkgs.system-sendmail}/bin/sendmail"'';
+    };
+
+    watchlist = lib.mkOption {
```

Comment:
Perhaps it would make sense to specify hooks per watchlist entry? (We can trivially multiplex them.)

Reply:
in my opinion, the user should handle this themselves in hook code by comparing `$WATCH_ITEM`. No need to add extra options to the module without clear use cases.